### ISSUE TITLE
EIP 1559 support

### ISSUE THREAD

**Issue by keeran** [EthWorks/ethereum.rb#171]
Hello all!

I've recently started experimenting with this library, along with ruby-eth, for generating transactions against contracts.

I'd like to follow the EIP 1559 standard and generate payloads with the appropriate fields present (`maxFeePerGas` & `maxPriorityFeePerGas`). I thought about forking both of the projects and hacking in the new fields, but if there is broader interest, it should probably become the standard in this / both projects?

I'm pretty new to the ecosystem, so I'm not sure how to approach testing (seems that the protocol specs are pretty dated too), so is there anyone who could help guide me on this mission?

Does it seem like a good idea to make the new standard... the standard in these libraries?

/cc @q9f (seems that you're pretty active here - please let me know if I can help. I'll try not to be too much of a burden :)

**Comment by q9f** [comment]
Hi, @keeran

I don't think this library will be maintained anymore. Your timing is not bad though, we created a "Ruby Ethereum" group on Telegram (join us) and I started to reimplement all Ethereum functionality with @chainoperator in https://github.com/q9f/eth.rb

We already replaced the broken ssl/crypto dependencies with the ruby secp256k1 library and implemented key pairs and addresses. @chainoperator is currently working on encryption/decryption while I'm hacking on sign/verify/recover.

Before creating a fork, let's collaborate? The transaction module would be next up on our checklist.

The checklist still lives in the old repository. https://github.com/se3000/ruby-eth/issues/68
